frontend/src/components/TagInput/TagInput.tsx

```diff
@@ -90,8 +90,17 @@ export const TagInput = forwardRef<TagInputProps, 'input'>(
     const addTag = useCallback(
       (event: SyntheticEvent, tag: string) => {
         if (event.isDefaultPrevented()) return
-        if (preventDuplicates && value.includes(tag)) return
-        onChange(value.concat(tag.split(',')))
+        if (preventDuplicates) {
+          if (value.includes(tag)) return
+          const existingTags = new Set(value)
+          const newTags = new Set(tag.split(','))
+          const uniqueTags = [...newTags].filter(
+            (tag) => !existingTags.has(tag),
+          )
```

Comment:
nit: might not need to filter, we can just add new values to the set and create a list at the end instead of filtering? might be cleaner. ordering will not be lost as sets are ordered in javascript.

```ts
onChange(Array.from(new Set([...value, ...tag.split(',')]))
```

Reply:
>ordering will not be lost as sets are ordered in javascript
Wow i didn't know that! Thank you!

Done, as requested

Reply:
Note that JavaScript sets are sorted in insertion order, not lexical, fyi, hehe

Reply:
swee, thanks!